### ISSUE TITLE
tails_files upgrade/re-run: existing file conflict with cp and set -e 

### ISSUE THREAD

**Issue by ageis** [freedomofpress/securedrop#1286]
If `install.sh` is executed a second time, the script will fail after the following message:
```
cp: ‘/home/amnesia/Desktop/document.desktop’ and ‘/live/persistence/TailsData_unlocked/dotfiles/Desktop/document.desktop’ are the same file
```

Also, this is more of an edge case, but if I manually clean up the desktop shortcuts in persistence (`/live/persistence/TailsData_unlocked/`) but forget to remove the symlinks in `~/Desktop`:
```
cp: not writing through dangling symlink ‘/home/amnesia/Desktop/document.desktop’
```

cp has a number of options that may be helpful in the first case. I think --update is probably the right one.

```
       -u, --update
              copy only when the SOURCE file is newer than the destination file or when the destination file is missing

       --remove-destination
              remove each existing destination file before attempting to open it (contrast with --force)

       -f, --force
              if an existing destination file cannot be opened, remove it and try again (this option is ignored when the -n option is also used)

       -n, --no-clobber
              do not overwrite an existing file (overrides a previous -i option)

```

**Comment by garrettr** [comment]
Isn't `--force` the simplest solution here? Don't we **always** want to copy the latest version of the `.desktop` shortcuts when tails_files is run? `--update` seems like it would achieve that too, but is potentially more fragile. What if there are clock problems that mess up timestamps at some point? This actually happens fairly often on Tails systems, at least temporarily.

**Comment by ageis** [comment]
@garrettr Do you think we should put an -f on all instances of copy, just to be safe, or only the ones that are prone to "same file" errors? It's possible that more files could be affected by this than we're aware of now.

**Comment by ageis** [comment]
@garrettr This issue has been addressed in https://github.com/freedomofpress/securedrop/pull/1287. The error was actually due to the symlink rather than a need for certain options on `cp`. We fixed it by copying the shortcuts from `$INSTALL_DIR` (~/.securedrop) instead of from `~/Desktop` (which is symlinked to persistence on an already-SecureDrop-customized system), which is as it should've been anyway.
